crate_universe/src/cli/splice.rs

```diff
@@ -55,12 +60,15 @@ pub fn splice(opt: SpliceOptions) -> Result<()> {
     let extra_manifests_manifest =
         ExtraManifestsManifest::try_from_path(opt.extra_manifests_manifest)?;
 
+    // Determine the splicing workspace
+    let temp_dir = tempfile::tempdir().context("Failed to generate temporary directory")?;
+    let splicing_dir = match &opt.workspace_dir {
+        Some(dir) => dir.clone(),
+        None => temp_dir.as_ref().to_path_buf(),
+    };
```

Comment:
We can avoid making the tempdir if we're not going to use it:

```suggestion
    // Determine the splicing workspace
    let temp_dir;
    let splicing_dir = match &opt.workspace_dir {
        Some(dir) => dir.clone(),
        None => {
            temp_dir = tempfile::tempdir().context("Failed to generate temporary directory")?;
            temp_dir.as_ref().to_path_buf()
        },
    };
```